website/guides/running-terraform-in-automation.html.md

```diff
@@ -0,0 +1,279 @@
+configuration.
+3. A human review that plan, to ensure it is acceptable.
+4. Terraform takes the actions described in the plan.
+
+Steps 1, 2 and 4 can be carried out using the familiar Terraform CLI commands,
+with some additional options:
+
+* `terraform init -input=false`
+* `terraform plan -out=tfplan -input=false`
+* `terraform apply -input=false`
+
+The `-input=false` option indicates that Terraform should not attempt to
+prompt for input, and instead expect all necessary values to be provided by
+either configuration files or the command line. It may therefore be necessary
+to use the `-var` and `-var-file` options on `terraform plan` to specify any
+variable values that would traditionally have been manually-entered under
+interactive usage.
+
+## Plan and Apply on different machines
+
+When running in an orchestration tool, it can be difficult or impossible to
+ensure that the `plan` and `apply` subcommands are run on the same machine,
+in the same directory, with all of the same files present.
+
+Running `plan` and `apply` on different machines requires some additional
+steps to ensure correct behavior. A robust strategy is as follows:
+
+* After `plan` completes, archive the entire working directory, including the
+  `.terraform` subdirectory created during `init`, and save it somewhere
+  where it will be available to the apply step. A common choice is as a
+  "build artifact" within the chosen orchestration tool.
+* Before running `apply`, obtain the archive created in the previous step
+  and extract it _at the same absolute path_. This re-creates everything
+  that was present after plan, avoiding strange issues where local files
+  were created during the plan step.
+
+Terraform currently makes some assumptions which must be accommodated by
+such an automation setup:
+
+* The saved plan file can contain absolute paths to child modules and other
+  data files referred to by configuration. Therefore it is necessary to ensure
+  that the archived configuration is extracted at an identical absolute path.
+  This is most commonly achieved by running Terraform in some sort of isolation,
+  such as a Docker container, where the filesystem layout can be controlled.
+* Terraform assumes that the plan will be applied on the same operating system
+  and CPU architecture as where it was created. For example, this means that
+  it is not possible to create a plan on a Windows computer and then apply it
+  on a Linux server.
+
+## Interactive Approval of Plans
+
+Another challenge with automating the Terraform workflow is the desire for an
+interactive approval step between plan and apply. To implement this robustly,
+it is important to ensure that either only one plan can be outstanding at a
+time or that the two steps are connected such that approving a plan passes
+along  enough information to the apply step to ensure that the correct plan is
+applied, as opposed to some later plan that also exists.
```

Comment:
Some folks might not know it's actually possible to do this, i.e. that `apply`'s positional argument can be a planfile, so I'd add an example here.